master/internal/api_project_intg_test.go

```diff
@@ -784,3 +784,78 @@ func TestGetProjectByID(t *testing.T) {
 	require.Equal(t, projectName, project.Name)
 	require.Equal(t, resp.Project.Id, project.Id)
 }
+
+func TestGetMetadataValues(t *testing.T) {
+	api, curUser, ctx := setupAPITest(t, nil)
+	_, projectIDInt := createProjectAndWorkspace(ctx, t, api)
+	projectID := int32(projectIDInt)
+	exp := createTestExpWithProjectID(t, api, curUser, int(projectID))
+
+	numRuns := 4
+	for i := 0; i < numRuns; i++ {
+		task := &model.Task{TaskType: model.TaskTypeTrial, TaskID: model.NewTaskID()}
+		require.NoError(t, db.AddTask(context.Background(), task))
+		require.NoError(t, db.AddTrial(context.Background(), &model.Trial{
+			State:        model.PausedState,
+			ExperimentID: exp.ID,
+			StartTime:    time.Now(),
+		}, task.TaskID))
+	}
+
+	resp, err := api.SearchRuns(ctx, &apiv1.SearchRunsRequest{ProjectId: &projectID})
+	require.NoError(t, err)
+
+	// Add metadata
+	rawMetadata1 := map[string]any{
+		"test_key": "test_value1",
+		"nested": map[string]any{
+			"nested_key": "nested_value1",
+		},
+	}
+	rawMetadata2 := map[string]any{
+		"test_key": "test_value1",
+		"nested": map[string]any{
+			"nested_key": "nested_value2",
+		},
+	}
+	rawMetadata3 := map[string]any{
+		"test_key": "test_value2",
+		"nested": map[string]any{
+			"nested_key": "nested_value2",
+		},
+	}
+	rawMetadata4 := map[string]any{
+		"test_key": "test_value3",
+		"nested": map[string]any{
+			"nested_key": "nested_value1",
+		},
+	}
+
+	rawMetadata := []map[string]any{}
+	rawMetadata = append(rawMetadata, rawMetadata1, rawMetadata2, rawMetadata3, rawMetadata4)
```

Comment:
nit (non-blocking): Would it be worth instead having each of the `rawMetadataX`'s variables instead be combined into the rawMetadata map like:
```
rawMetadata := []map[string]any{
    {
        "test_key": "test_value1", 
        "nested": map[string]any{"nested_key": "nested_value1"},   
    },
    ....
}
``` 

Or generated using a loop:
```
rawMetadata := []map[string]any{}
for i := range numRuns {
     metadataEntry := map[string]any{
          "test_key": fmt.Sprintf("test_value%d", i),
          "nested": map[string]any{"nested_key": fmt.Sprintf("nested_value%d", i)},
     }
     rawMetadata = append(rawMetadata,
}
```